Source/JavaScriptCore/runtime/JSPromiseAllGlobalContext.h

```diff
@@ -65,28 +44,30 @@ class JSPromiseAllGlobalContext final : public JSInternalFieldObjectImpl<JSPromi
 
     static Structure* createStructure(VM&, JSGlobalObject*, JSValue);
 
-    static JSPromiseAllGlobalContext* createWithInitialValues(VM&, Structure*);
-    static JSPromiseAllGlobalContext* create(VM&, Structure*, JSValue promise, JSValue values, JSValue remainingElementsCount);
+    static JSPromiseAllGlobalContext* create(VM&, JSValue promise, JSValue values, JSValue remainingElementsCount);
 
-    JSValue promise() const { return internalField(Field::Promise).get(); }
-    JSValue values() const { return internalField(Field::Values).get(); }
-    JSValue remainingElementsCount() const { return internalField(Field::RemainingElementsCount).get(); }
+    JSValue promise() const { return m_promise.get(); }
+    JSValue values() const { return m_values.get(); }
+    JSValue remainingElementsCount() const { return m_remainingElementsCount.get(); }
 
-    void setPromise(VM& vm, JSValue promise) { internalField(Field::Promise).set(vm, this, promise); }
-    void setValues(VM& vm, JSValue values) { internalField(Field::Values).set(vm, this, values); }
-    void setRemainingElementsCount(VM& vm, JSValue remainingElementsCount) { internalField(Field::RemainingElementsCount).set(vm, this, remainingElementsCount); }
+    void setPromise(VM& vm, JSValue promise) { m_promise.set(vm, this, promise); }
+    void setValues(VM& vm, JSValue values) { m_values.set(vm, this, values); }
+    void setRemainingElementsCount(VM& vm, JSValue remainingElementsCount) { m_remainingElementsCount.set(vm, this, remainingElementsCount); }
 
 private:
-    JSPromiseAllGlobalContext(VM& vm, Structure* structure)
+    JSPromiseAllGlobalContext(VM& vm, Structure* structure, JSValue promise, JSValue values, JSValue remainingElementsCount)
         : Base(vm, structure)
+        , m_promise(promise, WriteBarrierEarlyInit)
+        , m_values(values, WriteBarrierEarlyInit)
+        , m_remainingElementsCount(remainingElementsCount, WriteBarrierEarlyInit)
     {
     }
 
-    void finishCreation(VM&, JSValue promise, JSValue values, JSValue remainingElementsCount);
+    WriteBarrier<Unknown> m_promise;
+    WriteBarrier<Unknown> m_values;
+    WriteBarrier<Unknown> m_remainingElementsCount;
```

Comment:
question: `remainingElementsCount` is always number so can we make this just `uint64_t` ?